skills/projectCommand.js

```diff
@@ -5,25 +5,24 @@ const Actions = require('../lib/actions/actions.js');
 // const AddLinkToProjectResponse = require('../lib/actions/AddLinkToProjectResponse.js');
 
 
-module.exports = async function(controller) {
+module.exports = function(controller) {
   controller.on('slash_command', function (bot, message) {
     if(message.command === '/project'){
       // "find channel and excute action"
       console.log(message.channel_id)
-      if(message.channel_id === 'DK3H5J1B7') {
+      console.log(message.text)
+      if(message.channel_id === 'CK5B7E5FZ') {
         // bot.reply(message,"Channel ID")
         console.log(message.text)
-        let find = await Project.findByChannel(message.channel_id);
+        let find = Project.findByChannel(message.channel_id);
+        // this just takes text arguments (add goog google.com)
+        let action = Actions.fromMessageText(message.text);
+        let links = AddLinkToProjectAction.parseParams(action);
```

Comment:
Hey!
I think here we want to use `action.execute()` instead of `AddLinkToProjectAction.parseParams` .
The `action` on line 19 is a new instance of whichever `action` is coming in `fromMessageText`. Since right now it's probably `add`, then `action` is equal to ` new addLinkToProjectAction()`.. so we can say, `action.execute(params)`... in this case, `params` = `message.text`

Reply:
`action.execute` returns a new instance of `AddLinkToProjectResponse` which is a promise.. so that means when we run `action.execute(params)`, we’ll be `await`ing a response... 
but remember! we can only use `await` in `async` functions.. we can use that or resolve this in another way.

Reply:
100% agree with these comments. `AddLinkToProjectResponse` is a class so it isn’t itself a promise. Maybe it’s sendToBot is?

Reply:
right now, the AddLinkToProjectResponse class looks like this:
```
class AddLinkToProjectResponse {
  constructor(label, link){
    this.label = label;
    this.link = link;
  }

  sendToBot(bot) {
    return bot.reply('Link stored.')
  }
}```

at the moment, sendToBot() is not a promise.